tests/kuttl/tests/dataplane-service-custom-image/00-assert.yaml

```diff
@@ -29,7 +29,6 @@ status:
     reason: Ready
     status: "True"
     type: SetupReady
-  deployed: true
```

Comment:
This does not seem right.  Maybe we just need re-queue when all the roles are not created. Below piece of code before [1] after we process the existing items in roles.

```
if len(instance.Spec.Roles) > len(roles.Items) {
	shouldRequeue = true
}
```

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplane_controller.go#L208

Reply:
I don't understand, are you asking to move or to remove the code?

Reply:
If there are roles/nodes in the spec or not, I think expected final status of dataplane should be deployed: true. So you don't have to remove it from that assert. 

Also, I'm also suggesting to simplify the current code changes and move the check to L208.

Reply:
but if the reconcile loop isn't finding the roles, considering deployed = true isn't misleading?

Reply:
Roles and Nodes are both optional in the spec[1]. If there are roles in the spec like this kuttl test[2] and it can't find them, should not it requeue and wait for it.

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/openstackdataplane_types.go#L32-L35
[2] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/tests/kuttl/tests/dataplane-service-custom-image/00-dataplane-create.yaml#L24-L26



[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/openstackdataplane_types.go#L32-L35

Reply:
I've made some changes to the code, so deployed will stay true in this case